### PR TITLE
Pybind11: more python binding (especially for filedata) and adding fileIO example

### PR DESCRIPTION
**Pull request by weinmueller**
Add more pybind11 functions plus one example for reading and creating
xml files from python. 

Also fixing filedata output for boundary conditions.

# Please consider the following checklist before issuing a pull
request:
- [X] Have you added an explanation of what your changes do and why
  you'd like us to include them?
- [X] Have you documented any new codes using Doxygen comments?
- [X] Have you written new tests or examples for your changes?
-----

### CODE REVIEW

File: src/gsCore/gsBoundary.cpp [hunk 1]
```diff
@@ -285,6 +285,16 @@ void boundaryInterface::reorderCorners(gsMatrix<index_t> & boundary) const
...
             .value("northeast"     , boundary::northeast     )
             .export_values();
+
+        py::class_<boxSide>(m, "boxSide")
+            .def(py::init<short_t>())
+            .def("index", static_cast<short_t (boxSide::*)() const> (&boxSide::index), "Returns side index.");
+
+        py::class_<patchSide, boxSide>(m, "patchSide")
+            .def(py::init<index_t, boxSide>())
+            .def("patch", &patchSide::patchIndex, "Return the box side.")
```
Comment (stefantakacs) [inline src/gsCore/gsBoundary.cpp]: "patchIndex" + fix docs

File: src/gsCore/gsMultiPatch_.cpp [hunk 2]
```diff
@@ -15,8 +15,9 @@ namespace gismo
   void pybind11_init_gsMultiPatch(py::module &m)
   {
     using Base = gsFunctionSet<real_t>;
+    using BaseB = gsBoxTopology;
     using Class = gsMultiPatch<real_t>;
-    py::class_<Class,Base>(m, "gsMultiPatch")
+    py::class_<Class,Base, BaseB>(m, "gsMultiPatch")
```
Comment (stefantakacs) [inline src/gsCore/gsMultiPatch_.cpp]: using Base1 = gsFunctionSet<real_t>;
using Base2 = gsBoxTopology;

would be clearer than BaseB.

**Review by stefantakacs — APPROVED** [review]
OK.

Please change requested items first.

File: src/gsCore/gsMultiPatch_.cpp (hunk 2 above)
Comment (weinmueller) [reply src/gsCore/gsMultiPatch_.cpp]: This as well.

File: src/gsCore/gsBoundary.cpp (hunk 1 above)
Comment (weinmueller) [reply src/gsCore/gsBoundary.cpp]: Fixed.

**Review by filiatra — APPROVED** [review]

**Pull request merged**